test/jasmine/tests/gl2d_plot_interact_test.js

```diff
@@ -214,6 +214,27 @@ describe('Test gl plot side effects', function() {
         .catch(failTest)
         .then(done);
     });
+
+    it('@gl should fire *plotly_webglcontextlost* when on webgl context lost', function(done) {
+        var _mock = Lib.extendDeep({}, require('@mocks/gl2d_12.json'));
+
+        Plotly.plot(gd, _mock).then(function() {
+            return new Promise(function(resolve, reject) {
+                gd.on('plotly_webglcontextlost', resolve);
+                setTimeout(reject, 10);
+
+                var ev = new window.WebGLContextEvent('webglcontextlost');
+                var canvas = gd.querySelector('.gl-canvas-context');
```

Comment:
What I meant by "triggered on any of the canvases losing their context" is, for this one plot we make 3 canvases, each with its own webgl context... any of them could in principle lose its context, right?

```
> gd.querySelectorAll('canvas')
NodeList(3) [
    canvas.gl-canvas.gl-canvas-context,
    canvas.gl-canvas.gl-canvas-focus,
    canvas.gl-canvas.gl-canvas-pick
]
```

Reply:
Yeah, you caught me being lazy. 

Better tests in https://github.com/plotly/plotly.js/pull/2986/commits/905956a62560d92cfc5b7b92e93b1b9cb80da546

Reply:
even better tests in https://github.com/plotly/plotly.js/pull/2986/commits/3a38d739b7e33415e883f1599f062cd7c3163f8e